sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -391,7 +404,13 @@ public Builder catalogResolver(CatalogResolver catalogResolver)
       return this;
     }
 
-    public SqlTestFramework build()
+    public Builder withOverrideModule(Module m)
```

Comment:
could this method be considered to be added in `QueryComponentSupplier` rather? since that interface manages the guice configuration.
Also, its usages are done right after `configureGuice`

Reply:
I've kinda understood the reason for this change now - `QueryComponentSupplier` is per test class right now I think. :(

Reply:
yes this was needed to be able to change config per test method instead of per testclass